refarch-backend/src/main/java/de/muenchen/refarch/theentity/TheEntityService.java

```diff
@@ -0,0 +1,62 @@
+package de.muenchen.refarch.theentity;
+
+import de.muenchen.refarch.common.exception.NotFoundException;
+import de.muenchen.refarch.security.Authorities;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+@Transactional
+public class TheEntityService {
+    private static final String MSG_NOT_FOUND = "Could not find entity with id %s";
+
+    private final TheEntityRepository theEntityRepository;
+
+    @PreAuthorize(Authorities.HAS_AUTHORITY_READ_THEENTITY)
+    public TheEntity getTheEntity(final UUID theEntityId) {
+        log.info("Get TheEntity with ID {}", theEntityId);
+        return getEntityOrThrowException(theEntityId);
+    }
+
+    @PreAuthorize(Authorities.HAS_AUTHORITY_READ_THEENTITY)
+    public Page<TheEntity> getAllEntities(final int pageNumber, final int pageSize) {
+        log.info("Get AllEntities with at Page {} with a PageSize of {}", pageNumber, pageSize);
+        final Pageable pageRequest = PageRequest.of(pageNumber, pageSize);
+        return theEntityRepository.findAll(pageRequest);
+    }
+
+    @PreAuthorize(Authorities.HAS_AUTHORITY_WRITE_THEENTITY)
+    public TheEntity createTheEntity(final TheEntity entity) {
```

Comment:
You mean, instead of Request/Response DTOs we have DTOs for Post/Put/Get/.....? Thats a way we could do it, but in my opinion it should only be done if needed and not be the standard for everthing, because it would be a lot of boilerplate code.